test/fixtures/test-runner/global-setup-teardown/async-setup-teardown.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const fs = require('fs');
+
+const asyncFlagPath = process.env.ASYNC_FLAG_PATH;
+
+async function globalSetup() {
+  console.log('Async setup starting');
+
+  // Simulate an async operation
+  await new Promise((resolve) => setTimeout(resolve, 500));
+
+  fs.writeFileSync(asyncFlagPath, 'Setup part');
+  console.log('Async setup completed');
+}
+
+async function globalTeardown() {
+  console.log('Async teardown starting');
+
+  // Simulate an async operation
+  await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
Please use `setTimeout()` from `node:timers/promises` for this type of thing. Here and in other places.